### PR TITLE
Swoole 5

### PR DESCRIPTION
**Pull request by tillkruss**
Related #22.

The `swoole` builds are failing.

### CODE REVIEW

**Comment by kingIZZZY** [comment]
Do these build errors mean that only >= 20200930 can have swoole 5.0.0, whereas anything <= 20190902 must remain with latest 4.* ?

**Comment by tillkruss** [comment]
I don't know, I'm not familiar with compiling Swoole. Somehow it's not working in that Docker image and needs fixing as of v5. Maybe a system dependency.

**Comment by kingIZZZY** [comment]
- [Here by v5.0.0](https://github.com/swoole/swoole-src/releases/tag/v5.0.0) it seems to say **"Minimum PHP version requirement is 8.0"**
- [Here by v4.6.0](https://github.com/swoole/swoole-src/releases/tag/v4.6.0) **"No longer support PHP-7.1"**
- In the [README compilation instructions](https://github.com/swoole/swoole-src#compiling-requirements) it says **"Compiling requirements - PHP 7.2.0 or later (The higher the version, the better the performance.)"**
Isn't this the issue - the PHP version, rather than other dependencies or whatever?

Does it make sense at all for this heroku repository system to have one version for earlier PHP and another version for later PHP?
For example in this case we should have swoole-v4.8.11 (or whatever v4.* version it might be) for PHP < 8.0, and swoole-v5.0.0 for PHP >= 8.0 ?

**Comment by kingIZZZY** [comment]
I tried hacking together my theoretical solution: https://github.com/cachewerk/heroku-php-extensions/pull/22/commits/4f321af739520e5385fa5e4c535385c67fa52fd9
But I still can't build without the env / secrets or whatever..

Can you try building this last commit in my PR #22 ?

**Comment by tillkruss** [comment]
> Does it make sense at all for this heroku repository system to have one version for earlier PHP and another version for later PHP? For example in this case we should have swoole-v4.8.11 (or whatever v4.* version it might be) for PHP < 8.0, and swoole-v5.0.0 for PHP >= 8.0 ?

I think we should do Swoole 5 on the Heroku-22 stack only.

**Comment by kingIZZZY** [comment]
OK lemme try to push a commit to that effect

I'm not sure why swoole 5 can't also work for heroku-18 and heroku-20... but let's try for now just heroku-22

**Comment by kingIZZZY** [comment]
I pushed in my PR #22 https://github.com/cachewerk/heroku-php-extensions/pull/22/commits/9dcf86df96a168ef37533171dd70ce0230dfdb06
Either way my builds always fail due to lacking env / secrets
It's possible it might even work for heroku-18 and heroku-20... if you can try to run my PRs please? 🙏

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -121,8 +123,21 @@ jobs:
             ${{ matrix.stack }} deploy.sh --overwrite \
             extensions/no-debug-non-zts-${{ matrix.series }}/relay-0.4.4
 
-      - name: Build swoole extension
-        if: matrix.stack != 'heroku-22' || matrix.series >= 20210902
+      - name: Build swoole 4.x extension
+        # skip build on heroku-22
+        if: matrix.stack != 'heroku-22'
```
Comment (kingIZZZY) [inline .github/workflows/build.yml:128]: So there will be no swoole 4.x neither swoole 5.x for someone on heroku-22 with PHP 8.0 ?

**Comment by kingIZZZY** [comment]
Well congrats at least it builds
For the time being I personally just need swoole 5 for heroku-22 on PHP 8.1, so this'll work for me. (Anyone on heroku-22 with PHP 8.0 though... 🤷‍♂️)

How soon can this be released so it is usable on heroku apps?

**Comment by tillkruss** [comment]
`heroku-22` is PHP 8.1+

File: .github/workflows/build.yml (hunk 1 above)
Comment (tillkruss) [reply .github/workflows/build.yml:128]: Have you tried using 8.0 on `heroku-22`?

Comment (kingIZZZY) [reply .github/workflows/build.yml:128]: yep sorry I realized later your other comment - heroku-22 is only php >= 8.1

**Comment by tillkruss** [comment]
Closing for #30.

**Pull request closed without merging**